src/mds/Mutation.h

```diff
@@ -276,6 +277,8 @@ struct MDRequestImpl : public MutationImpl {
   // indicates how may retries of request have been made
   int retry;
 
+  bool is_batch_head = false;
```

Comment:
maybe change this to 
bool is_batch_head() const {  return !batch_head; }

Reply:
@ukernel when the MDRequest has not reach the point that add batch_ops, is_batch_head is set to false while batch_head is also empty. It's necessary for is_batch_head to be false at this circumstance when deciding whether to iterate all batch_ops in MDCache::request_forward method which could be invoked before the MDRequest's batch_head is set. So, it seems better for is_batch_head to remain a independent field:-)